x/dualstaking/README.md

```diff
@@ -0,0 +1,138 @@
+For every validator delegation exists a parralel provider delegation and vice versa.
+When a provider stakes his tokens, he is a self delegating provider and delegating to a validator with the same amount. This give him gov influence and support a validator of his choosing which in turn boosts the security of the chain (this is also true to providers delegators).
+The same way happens with validators, when a validator stakes his tokens (or a delegator), using hooks a delegation is created to an empty provider. Than using the dualstaking module he can redelegate from the empty provider to an actual provider.
+Since the module uses hooks to achieve this functionality it does not need to move coins, the staking module does it for him.
+For a specific address, at all times the amount of validators delegation is equal to providers delegations.
+The following are use cases of the dualstaking module:
+
+#### Validator delegation
+
+1. Call delegate method of the staking module.
+2. Hook on create delegation and delegate to the empty provider.
+
+#### Validator unbonding
+
+1. Call unbond method of the staking module.
+2. Hook on create delegation and unbond the same amount from the providers delegations uniformaly with priority to the empty provider.
+
+#### Validator slashing
+
+1. Hook on calidator slashing
+2. remove the slashed amount uniformaly from the providers delegations
+
+#### Provider delegation
+
+1. Call delegate method of the dualstaking module.
+2. Hook on create delegation and delegate to the empty provider.
+3. redelegate from the empty provider to the actual provider.
+
+#### Provider unbonding
+
+1. redelegate from the the provider to the empty provider.
+2. Call unbond method of the dualstaking module.
+3. Hook on create delegation and unbond from empty provider.
+
+### Hooks
+
+Dual staking module uses [staking hooks](keeper/hooks.go) to achieve its functionality.
+1. AfterDelegationModified: this hook is called whenever a delegation is changed, whether it is created, or modified (NOT when completly removed). it calculates the difference in providers and validators stake to determine the action of the user (delegation or unbonding) depending on who is higher and than does the same with provider delegation.
+    * If provider delegations > validator delegations: user unbonded, uniform unbond from providers delegations (priority to empty provider)
+    * If provider delegations < validator delegations: user delegation, delegate to the empty provider
+2. BeforeDelegationRemoved: this hook is called when a delegation to a validator is removed (unbonding of all the tokens). uniform unbond from providers delegations
+3. BeforeValidatorSlashed: this hook is called when a validator is being slashed. to make sure the balance between validator and provider delegation is kept it uniform unbond from providers delegations the slashed amount. 
+
+### RedelegateFlag
```

Comment:
instead of making a subchapter with too much info (in my opinion), you can add the following sentence in the end of the "Hooks" chapter and remove this whole chapter. Sentence to add:
"Note, in case of a validator redelegation, the hooks are disabled"

Also, when you remove this chapter, don't forget to take it off from the table of contents

Reply:
this chapter is important to, we want to document the antehandler and how we make sure to ignore the redelegate message.
it is not trivial

Reply:
ok.

replace the paragraph with this (AI phrasing):

To prevent the dual staking module from taking action in the case of validator redelegation, we utilize the [antehandler](ante/ante_handler.go). When a redelegation message is being processed, the RedelegateFlag is set to true, and the hooks will disregard any delegation changes. It is important to note that the RedelegateFlag is stored in memory and not in the chain’s state.

Reply:
ok.

replace the paragraph with this (AI phrasing):

To prevent the dual staking module from taking action in the case of validator redelegation, we utilize the [antehandler](https://github.com/lavanet/lava/pull/ante/ante_handler.go). When a redelegation message is being processed, the RedelegateFlag is set to true, and the hooks will disregard any delegation changes. It is important to note that the RedelegateFlag is stored in memory and not in the chain’s state.